src/plugins/data/common/search/es_search/types.ts

```diff
@@ -37,12 +37,13 @@ export type ISearchRequestParams<T = Record<string, any>> = {
   trackTotalHits?: boolean;
 } & Search<T>;
 
-export interface IEsSearchRequest extends IKibanaSearchRequest {
-  params?: ISearchRequestParams;
+export interface IEsSearchRequest<T = Record<string, any>, P = ISearchRequestParams<T>>
+  extends IKibanaSearchRequest {
+  params?: P;
```

Comment:
This allows us to override `params` for `EqlSearchStrategyRequest` below to be those required by the EQL search endpoint.